sdk/cosmosdb/cosmos/src/ClientContext.ts

```diff
@@ -273,6 +273,55 @@ export class ClientContext {
     }
   }
 
+  public async patch<T>({
+    body,
+    path,
+    resourceType,
+    resourceId,
+    options = {},
+    partitionKey
+  }: {
+    body: any;
+    path: string;
+    resourceType: ResourceType;
+    resourceId: string;
+    options?: RequestOptions;
+    partitionKey?: PartitionKey;
+  }): Promise<Response<T & Resource>> {
+    try {
+      const request: RequestContext = {
+        globalEndpointManager: this.globalEndpointManager,
+        requestAgent: this.cosmosClientOptions.agent,
+        connectionPolicy: this.connectionPolicy,
+        method: HTTPMethod.patch,
+        client: this,
+        operationType: OperationType.Patch,
+        path,
+        resourceType,
+        body,
+        resourceId,
+        options,
+        plugins: this.cosmosClientOptions.plugins,
+        partitionKey
+      };
+
+      request.headers = await this.buildHeaders(request);
+      this.applySessionToken(request);
+
+      // replace will use WriteEndpoint since it uses PUT operation
+      request.endpoint = await this.globalEndpointManager.resolveServiceEndpoint(
+        request.resourceType,
+        request.operationType
+      );
+      const response = await executePlugins(request, executeRequest, PluginOn.operation);
+      this.captureSessionToken(undefined, path, OperationType.Replace, response.headers);
```

Comment:
shouldn't operation type be patch?

Reply:
yep these are copypasta since this method is the same, let me fix